src/NServiceBus.MessagingBridge/AddressMap.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NServiceBus;
+using NServiceBus.Raw;
+using NServiceBus.Transport;
+
+/// <summary>
+/// How to map an address depends upon which transport the message is being sent to, 
+/// so it is necessary to have an address mapping dictionary for each transport
+/// </summary>
+class AddressMap(IReadOnlyDictionary<string, IStartableRawEndpoint> dispatchers) : IAddressMap
+{
+    readonly Dictionary<string, TransportAddressMap> addressMap =
+        dispatchers.ToDictionary(dispatcher => dispatcher.Key, _ => new TransportAddressMap());
+
+    readonly HashSet<string> endpoints = [];
+
+    public void Add(BridgeTransport transport, BridgeEndpoint endpoint)
+    {
+        if (!endpoints.Add(endpoint.Name))
+        {
+            throw new InvalidOperationException($"{endpoint.Name} has already been added to the address map.");
+        }
+
+        foreach (var targetTransport in addressMap.Keys)
+        {
+            var queueAddress = new QueueAddress(endpoint.Name);
+            var targetAddress = targetTransport == transport.Name
+                ? endpoint.QueueAddress ?? dispatchers[targetTransport].ToTransportAddress(queueAddress)
```

Comment:
could this and the logic for `sourceAddress` generation be extracted - these seem to identical.